src/qililab/instruments/qblox/qblox_module.py

```diff
@@ -253,9 +253,24 @@ def _generate_program(  # pylint: disable=too-many-locals
                     wait_time=int(wait_time),
                 )
             )
-        self._append_acquire_instruction(
-            loop=bin_loop, bin_index=bin_loop.counter_register, sequencer_id=sequencer, weight_regs=weight_registers
-        )
+            if pulse_bus_schedule.port == "feedline_input":
+                self._append_acquire_instruction(
+                    loop=bin_loop,
+                    bin_index=bin_loop.counter_register,
+                    sequencer_id=sequencer,
+                    weight_regs=weight_registers,
+                    acq_index=i,
+                )
```

Comment:
Do we know with certainty that port is named `feedline_input`? What if the user provides a different name in the runcard?

Reply:
should also work with `if self.module_type == "QRM":` do you think it's better?

Reply:
I think this is better since "QRM" can't be changed by the user right? Another option is to check if controller is ReadoutController but I don't know if we have access from this class.